.travis/travis.py

```diff
@@ -16,95 +49,198 @@ def test_diffs(diffs):
+                g_upstream_repo = g.get_repo(
+                    full_name_or_id=GIT_UPSTREAM_REPO_SLUG)
+
+                # Define common attributes for new PR
+                pr_branch_name = (
+                    '/').join([HUB_REPO, HUB_RELEASE, HUB_OS_NAME, HUB_OS_CODE_NAME])
+                pr_head_name = GIT_USER + ':' + pr_branch_name
+
+                pr_remote = git.remote.Remote(repo=repo, name='origin')
+                pr_remote.add(repo=repo, name='upstream_pr',
+                              url=GIT_REMOTE_ORIGIN_TOKEN_URL)
+
+                # Commit changes to Dockerfiles
+                repo.git.add(all=True)
+                message = "Updating Dockerfiles\n" + \
+                    "This is an automated CI commit"
+                repo.git.commit(m=message, author=GIT_AUTHOR)
+
+                # Update the Docker Library
+                manifest = os.path.join(hub_repo_dir, 'manifest.yaml')
+                output = os.path.join(hub_repo_dir, HUB_REPO)
+                create_dockerlibrary = import_create_dockerlibrary(
+                    hub_repo_dir)
+                create_dockerlibrary.main((
+                    '--manifest', manifest,
+                    '--output', output))
+
+                # Commit changes to Docker Library
+                repo.git.add(all=True)
+                message = "Updating Docker Library\n" + \
+                    "This is an automated CI commit"
+                repo.git.commit(m=message, author=GIT_AUTHOR)
+
+                # Create new branch from current head
+                new_branch = repo.create_head(pr_branch_name)
+
+                try:
+                    # Check if branch already exists
+                    g_branch = g_origin_repo.get_branch(branch=pr_branch_name)
+                    try:
+                        repo.git.push(
+                            'upstream_pr', pr_branch_name + ':' + pr_branch_name, force=True)
+                    except Exception as inst:
+                        inst.stderr = None
+                        raise ValueError(
+                            ("Force push to branch:{branch} failed! "
+                             "Stderr ommited to protect secrits.").format(branch=pr_branch_name))
+
+                except github.GithubException as exception:
+                    # Continue if branch does not already exist
+                    if exception.data['message'] == "Branch not found":
+                        try:
+                            repo.git.push(
+                                'upstream_pr', pr_branch_name + ':' + pr_branch_name, u=True)
+                        except Exception as inst:
+                            inst.stderr = None
+                            raise ValueError(
+                                ("Set-upstream push to branch:{branch} failed! "
+                                 "Stderr ommited to protect secrits.").format(branch=pr_branch_name))
```

Comment:
what happens if the error message is NOT "Branch not found", I think it should raise the exception instead of continuing isn't it?

Reply:
Fixed